### DISCUSSION TITLE
How to change width of menu when using vertical menu

### DISCUSSION THREAD

**Discussion by seoBOXX** [templaza/astroid-framework#199] [Q&A] [answered]
Hi there, 

i am using the vertical menu left sided on my dev-platform on https://joomla4.land-dinslaken.de. Is there any way to adjust the left menu width to fit the menu items?

Kind regards,
Andreas

**Comment by Chacapamac** [comment]
@seoBOXX 
Did you try the Astroid Template Manager?
![of-canvas](https://user-images.githubusercontent.com/1907276/192290923-92ec8f8e-8f2e-4143-b0eb-a6fa77618b2d.jpg)

**Reply by Chacapamac** [reply]
Or you maybe can use
```
ul.astroid-sidebar-menu {
white-space: nowrap;
}
```

**Reply by seoBOXX** [reply]
The CSS wont work for me because there are subitems in den menu, wich require even more width. I have no settings in the Off-Canvas menu section. I created a module with the menu and the mod-pos offcanvas.

**Reply by seoBOXX** [reply]
![Screenshot 26-09-2022 15-52-24 Uhr](https://user-images.githubusercontent.com/33403615/192294474-b5c36373-aa2d-4982-85a7-46c980a76150.jpg)
![Screenshot 26-09-2022 15-52-53 Uhr](https://user-images.githubusercontent.com/33403615/192294477-fb03b37c-3e7e-4db6-99a5-6c7d3c8460cb.jpg)

**Reply by Chacapamac** [reply]
> I have no settings in the Off-Canvas menu section.

**Ok, this is not correct, You should see the options????**

Joomla version?
What Astroid Framework and template version are you using?
What Template are you using?

**It seem that you are missing something in the Astroid installation?**
Did you try to reinstall the latest **Framework/Template** over your installation, something is missing here.

**Do a backup First, but it should fix this.**
Try Akeeba Backups — Top Joomla Free Component for Backup _(everybody use this...)_
https://www.akeeba.com/products/akeeba-backup.html

**New Astroid Version**
https://github.com/templaza/astroid-framework/releases/tag/v2.5.20

**Reply by seoBOXX** [reply]
Joomla Version 4.2.2
Astroid Framework Version 2.5.20
Template Version 2.5.19
Astroid Template Zero Style

I don't see this options. I already use AkeebaBackup (everytime my first choice). Reinstall Astroid Framework did not change anything.

**Reply by seoBOXX** [reply]
I completly removed Astroid and reinstalled ... no changes for me ... any ideas?

**Reply by Chacapamac** [reply]
Weird, This is what you should see in the Joomla 4 Version 4.2.2 > Extensions Manage > search input > astroid...
![astroid-](https://user-images.githubusercontent.com/1907276/192635832-20a28ce3-dcdc-4f83-bdd6-2f7da73c5e47.jpg)

If you reinstall, something is bugging Astroid interface  as you should see tons of options under **Sticky Header**, **Off Canvas Menu** and **Animation**

As exemple you should see the image in my precedent image/post for  **Off Canvas Menu** and I add the **Sticky Header options** here.
![sticky-header](https://user-images.githubusercontent.com/1907276/192638309-2913595f-d2a1-40b6-a7cb-4343304d685e.jpg)

**Open the error console of your Browser and check for error**
![errors-console](https://user-images.githubusercontent.com/1907276/192638945-cb933974-ccf2-49a8-a94b-0826b40ac844.jpg)

**My guess is an extension that influx on the administration look or performance (ajax) is causing the problem???**

**Also you can check what is on or off in the Astroid plugin:**
![plugins](https://user-images.githubusercontent.com/1907276/192639399-dd01576f-bc62-4d17-92f5-909d3711fe64.jpg)




@sonvnn  I mark it as a BUG

**Comment by seoBOXX** [comment]
Hi and good morning from Germany,

i check it and i changed some of the PlugIn Options ... but still no changes. Would you be so kind and take a closer look at it? I will grant you admin rights and access. I really have no idea whats the problem.

**Reply by Chacapamac** [reply]
Yes, sure, send me your login infos at my personal e-mail —> p.toulze@grafcomm.ca, I will take a look
Questions:
**• Is this a live website or a test website?
• Did you make a Backup, just in case?**

**Reply by seoBOXX** [reply]
Hi and good morning, i send you an email with the needes infos. Thanks a lot and have a great day.

**Comment by sonvnn** [comment, marked as answer]
@Chacapamac I think he want to refer to Sidebar Menu in Sidebar layout not Off-canvas Menu.
@seoBOXX You can use custom css to override style
```
.astroid-header.has-sidebar { width: 360px; }
@media (min-width: 1201px) {
  body.header-sidebar-left {
      margin-left: 360px;
  }
}

```
You can change 360px to your desired value

Best Regards,
Sonny

**Reply by seoBOXX** [reply]
Now i am a bit confused ... whats the difference between sidebar and offcanvas menu?

**Reply by Chacapamac** [reply]
**Yes, Sonny is right, I make a mistake, sorry!**
The **Off-Canvas Menus** is the menu you see when you viewing the website on smaller device _(with the hamburger menu)_
The settings I show above are for the Off Canvas Mobile menu not the **SideBar** (Header Mode)

**OFF_CANVAS -----------------------------------**

![Screen Shot 2022-09-29 at 9 55 20 AM](https://user-images.githubusercontent.com/1907276/193050883-d3f9e291-b264-4813-a6b6-64a6f01f5f60.jpg)



**SIDEBAR  -----------------------------------**

![Screen Shot 2022-09-29 at 9 47 35 AM](https://user-images.githubusercontent.com/1907276/193050231-3ad98d50-0ad5-4d75-8221-8b77fee03864.jpg)


**SUGGESTION  -----------------------------------**
**I suggest @sonvnn  to add a setting in Astroid admin to set a desired with for the sidebar. Like that users can easily set their width without resorting to css**

@seoBOXX  — IF you still have the problem of your administration not showing the options, yo can still send me a email for me to check. I didn’t receive it yet. —>  p.toulze@grafcomm.ca

**Comment by seoBOXX** [comment]
@Chacapamac -> I sent the EMail this morning ... maybe it was going into spam but anyway, adding the css code is working now for me.

@sonvnn -> Thanks a lot ... CSS code works like a charm. as Chacapamac suggested a setting fot this would be helpfull :-)

Thanks a lot, stay healthy and take care,

Andy

**Reply by Chacapamac** [reply]
Yes, It was in my spam... I try the username/Password — It didn’t work, can you verify?
I use the normal Login Button, not the Web Authentication...

**Reply by seoBOXX** [reply]
The CSS worked great ... i resized the sidebar and everything is ok for me :-) Do the offcanvas settings have to appear anyway? If so please take a look. I tried also ... forgot to set you to super admin ... tested and login works now.

**Comment by Chacapamac** [comment]
@seoBOXX , thanks, I was able to login and I find the problem, it is nothing bad but @sonvnn  will have to take a look.

**THE PROBLEM**
![prob](https://user-images.githubusercontent.com/1907276/193414287-cb8514b8-e5de-408d-a26b-affb25aa0035.jpg)

**THE CAUSE**
![missing](https://user-images.githubusercontent.com/1907276/193414474-9e456995-e9fe-425f-b67e-785c7ea96bda.jpg)


The only thing I try is to reinstall over yours the last same Astroid framework and templates package (2.5.20) just to see if a file was missing or damaged. 
**Didn’t work**

• Curiously on an identical setup (only I have php 8.0.8 / You have 8.1.9, my options appear correctly (without the inline class “ng-hide”)

**Reply by Chacapamac** [reply]
If Sonny doesn't respond because the question is no mark as “Answer” Take my pictures and resend it as a bug.
Good Day. For now you can try to make the options show by css if you need to use them until its resolve.

**Good Day Everybody!**

**Reply by seoBOXX** [reply]
Thanks a lot, have a nice weekend, stay healthy and take care. Maybe you can give @sonvnn the backend access i gave you if needed.

**Comment by Chacapamac** [comment]
No problem my friends

Good Weekend !

Patrick
